### ISSUE TITLE
Issue with navigation and rendering of templates

### ISSUE THREAD

**Issue by trippo80** [OnsenUI/OnsenUI#170]
Hi,

I've got a new issue that I only can detect on slower devices, for example iPhone 4. 

I'm using templates with:
```
<script id="foobar.html" type="text/ng-template"></script>
```

And loading them into my navigator:
```
<ons-navigator var="theNavigator" page="foobar.html"></ons-navigator>
```

I then push pages with:
```
<button ng-click="theNavigator.pushPage('foobar.second.html')">Click me</button>
```

In each template I have a ons-toolbar, but on the second template I've added an additional class to that ons-toolbar which changes it's background. That class is added by watching a variable in the scope:
```
<ons-toolbar class="the-navigator" bo-class="item_category">
```

But the thing is, on this slower device the page transition is completed before the template has compiled/rendered so that the content is empty when the transition occurs and also the additional class for the toolbar is added after the transition which makes the background color animate to the new value. 

TLDR; Navigator page transition is performed before templates are compiled/rendered which affects slower devices (for example iPhone 4). On desktop this isn't an issue because the template rendering/compilation is completed before the transition.
